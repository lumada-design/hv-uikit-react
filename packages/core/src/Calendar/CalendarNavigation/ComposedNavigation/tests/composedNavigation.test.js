import React from "react";
import { mount } from "enzyme";
import HvProvider from "../../../../Provider";

import { HvComposedNavigation } from "../../index";

describe("<Navigation />", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = mount(
      <HvProvider>
        <HvComposedNavigation locale="en" visibleYear={2020} visibleMonth={4} />
      </HvProvider>
    );
  });

  it("should render correctly", () => {
    expect(wrapper.find(HvComposedNavigation)).toMatchSnapshot();
  });
});
