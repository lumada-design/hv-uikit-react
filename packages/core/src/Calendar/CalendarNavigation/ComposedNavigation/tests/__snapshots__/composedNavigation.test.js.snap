// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render correctly 1`] = `
<WithStyles(HvComposedNavigation)
  locale="en"
  visibleMonth={4}
  visibleYear={2020}
>
  <HvComposedNavigation
    classes={
      Object {
        "navigationContainer": "HvComposedNavigation-navigationContainer",
        "navigationMonth": "HvComposedNavigation-navigationMonth",
      }
    }
    locale="en"
    visibleMonth={4}
    visibleYear={2020}
  >
    <div
      className="HvComposedNavigation-navigationContainer"
    >
      <div
        className="HvComposedNavigation-navigationMonth"
      >
        <WithStyles(Navigation)
          className="HvComposedNavigation-navigationMonth"
          navigationText="April"
          onNavigateNext={[Function]}
          onNavigatePrevious={[Function]}
          onTextClick={[Function]}
        >
          <Navigation
            className="HvComposedNavigation-navigationMonth"
            classes={
              Object {
                "disabled": "HvCalendarNavigation-disabled",
                "icon": "HvCalendarNavigation-icon",
                "root": "HvCalendarNavigation-root",
                "text": "HvCalendarNavigation-text",
                "textWithoutHover": "HvCalendarNavigation-textWithoutHover",
              }
            }
            navigationText="April"
            onNavigateNext={[Function]}
            onNavigatePrevious={[Function]}
            onTextClick={[Function]}
          >
            <div
              className="HvCalendarNavigation-root"
            >
              <DropLeftXS
                className="HvCalendarNavigation-icon"
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <HvIconBase
                  className="HvCalendarNavigation-icon"
                  iconSize="XS"
                  name="DropLeftXS"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  tabIndex={0}
                >
                  <div
                    className="HvCalendarNavigation-icon HvIconBase-root HvIconBase-xs"
                    name="DropLeftXS"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    tabIndex={0}
                  >
                    <svg
                      focusable={false}
                      height={12}
                      viewBox="0 0 12 12"
                      width={12}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        className="color0"
                        d="M3.35 5.99994l4.6-4.6.7.7L4.75 6l3.9 3.9-.7.7z"
                        fill="#414141"
                      />
                    </svg>
                  </div>
                </HvIconBase>
              </DropLeftXS>
              <div
                className="HvCalendarNavigation-text"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="presentation"
                tabIndex={0}
              >
                <WithStyles(ForwardRef)
                  variant="normalText"
                >
                  <ForwardRef
                    classes={
                      Object {
                        "3xlTitle": "HvTypography-3xlTitle",
                        "4xlTitle": "HvTypography-4xlTitle",
                        "5xlTitle": "HvTypography-5xlTitle",
                        "disabledText": "HvTypography-disabledText",
                        "highlightText": "HvTypography-highlightText",
                        "lTitle": "HvTypography-lTitle",
                        "link": "HvTypography-link",
                        "mTitle": "HvTypography-mTitle",
                        "noWrap": "HvTypography-noWrap",
                        "normalText": "HvTypography-normalText",
                        "placeholderText": "HvTypography-placeholderText",
                        "root": "HvTypography-root",
                        "sTitle": "HvTypography-sTitle",
                        "sectionTitle": "HvTypography-sectionTitle",
                        "selectedNavText": "HvTypography-selectedNavText",
                        "vizText": "HvTypography-vizText",
                        "vizTextDisabled": "HvTypography-vizTextDisabled",
                        "xlTitle": "HvTypography-xlTitle",
                        "xsInlineLink": "HvTypography-xsInlineLink",
                        "xsTitle": "HvTypography-xsTitle",
                        "xxlTitle": "HvTypography-xxlTitle",
                        "xxsTitle": "HvTypography-xxsTitle",
                      }
                    }
                    variant="normalText"
                  >
                    <p
                      className="HvTypography-root HvTypography-normalText"
                    >
                      April
                    </p>
                  </ForwardRef>
                </WithStyles(ForwardRef)>
              </div>
              <DropRightXS
                className="HvCalendarNavigation-icon"
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <HvIconBase
                  className="HvCalendarNavigation-icon"
                  iconSize="XS"
                  name="DropRightXS"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  tabIndex={0}
                >
                  <div
                    className="HvCalendarNavigation-icon HvIconBase-root HvIconBase-xs"
                    name="DropRightXS"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    tabIndex={0}
                  >
                    <svg
                      focusable={false}
                      height={12}
                      viewBox="0 0 12 12"
                      width={12}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        className="color0"
                        d="M4.05 10.6l-.7-.7L7.25 6l-3.9-3.9.7-.7L8.65 6z"
                        fill="#414141"
                      />
                    </svg>
                  </div>
                </HvIconBase>
              </DropRightXS>
            </div>
          </Navigation>
        </WithStyles(Navigation)>
      </div>
      <WithStyles(Navigation)
        navigationText="2020"
        onNavigateNext={[Function]}
        onNavigatePrevious={[Function]}
      >
        <Navigation
          classes={
            Object {
              "disabled": "HvCalendarNavigation-disabled",
              "icon": "HvCalendarNavigation-icon",
              "root": "HvCalendarNavigation-root",
              "text": "HvCalendarNavigation-text",
              "textWithoutHover": "HvCalendarNavigation-textWithoutHover",
            }
          }
          navigationText="2020"
          onNavigateNext={[Function]}
          onNavigatePrevious={[Function]}
        >
          <div
            className="HvCalendarNavigation-root"
          >
            <DropLeftXS
              className="HvCalendarNavigation-icon"
              onClick={[Function]}
              onKeyDown={[Function]}
              tabIndex={0}
            >
              <HvIconBase
                className="HvCalendarNavigation-icon"
                iconSize="XS"
                name="DropLeftXS"
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <div
                  className="HvCalendarNavigation-icon HvIconBase-root HvIconBase-xs"
                  name="DropLeftXS"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  tabIndex={0}
                >
                  <svg
                    focusable={false}
                    height={12}
                    viewBox="0 0 12 12"
                    width={12}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      className="color0"
                      d="M3.35 5.99994l4.6-4.6.7.7L4.75 6l3.9 3.9-.7.7z"
                      fill="#414141"
                    />
                  </svg>
                </div>
              </HvIconBase>
            </DropLeftXS>
            <div
              className="HvCalendarNavigation-textWithoutHover"
              role="presentation"
              tabIndex={-1}
            >
              <WithStyles(ForwardRef)
                variant="normalText"
              >
                <ForwardRef
                  classes={
                    Object {
                      "3xlTitle": "HvTypography-3xlTitle",
                      "4xlTitle": "HvTypography-4xlTitle",
                      "5xlTitle": "HvTypography-5xlTitle",
                      "disabledText": "HvTypography-disabledText",
                      "highlightText": "HvTypography-highlightText",
                      "lTitle": "HvTypography-lTitle",
                      "link": "HvTypography-link",
                      "mTitle": "HvTypography-mTitle",
                      "noWrap": "HvTypography-noWrap",
                      "normalText": "HvTypography-normalText",
                      "placeholderText": "HvTypography-placeholderText",
                      "root": "HvTypography-root",
                      "sTitle": "HvTypography-sTitle",
                      "sectionTitle": "HvTypography-sectionTitle",
                      "selectedNavText": "HvTypography-selectedNavText",
                      "vizText": "HvTypography-vizText",
                      "vizTextDisabled": "HvTypography-vizTextDisabled",
                      "xlTitle": "HvTypography-xlTitle",
                      "xsInlineLink": "HvTypography-xsInlineLink",
                      "xsTitle": "HvTypography-xsTitle",
                      "xxlTitle": "HvTypography-xxlTitle",
                      "xxsTitle": "HvTypography-xxsTitle",
                    }
                  }
                  variant="normalText"
                >
                  <p
                    className="HvTypography-root HvTypography-normalText"
                  >
                    2020
                  </p>
                </ForwardRef>
              </WithStyles(ForwardRef)>
            </div>
            <DropRightXS
              className="HvCalendarNavigation-icon"
              onClick={[Function]}
              onKeyDown={[Function]}
              tabIndex={0}
            >
              <HvIconBase
                className="HvCalendarNavigation-icon"
                iconSize="XS"
                name="DropRightXS"
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <div
                  className="HvCalendarNavigation-icon HvIconBase-root HvIconBase-xs"
                  name="DropRightXS"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  tabIndex={0}
                >
                  <svg
                    focusable={false}
                    height={12}
                    viewBox="0 0 12 12"
                    width={12}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      className="color0"
                      d="M4.05 10.6l-.7-.7L7.25 6l-3.9-3.9.7-.7L8.65 6z"
                      fill="#414141"
                    />
                  </svg>
                </div>
              </HvIconBase>
            </DropRightXS>
          </div>
        </Navigation>
      </WithStyles(Navigation)>
    </div>
  </HvComposedNavigation>
</WithStyles(HvComposedNavigation)>
`;
